src/components/AwsProvisionerHealthTable/index.jsx

```diff
@@ -0,0 +1,296 @@
+    drawerOpen: false,
+  };
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  createSortedHealth = memoizeWith(
+    (healthData, sortBy, sortDirection) => {
+      const ids = sorted(healthData.running);
+
+      return `${ids.join('-')}-${sortBy}-${sortDirection}`;
+    },
+    (healthData, sortBy, sortDirection) => {
+      const healthSummary = {};
+      const { requestHealth, terminationHealth, running } = healthData;
+      const identifier = item =>
+        `${item.az}-${item.region}-${item.instanceType}`;
+      const or0 = defaultTo(0);
+
+      requestHealth.forEach(item => {
+        healthSummary[identifier(item)] = {
+          ...healthSummary[identifier(item)],
+          ...item,
+        };
+      });
+      terminationHealth.forEach(item => {
+        healthSummary[identifier(item)] = {
+          ...healthSummary[identifier(item)],
+          ...item,
+        };
+      });
+      running.forEach(item => {
+        healthSummary[identifier(item)] = {
+          ...healthSummary[identifier(item)],
+          ...item,
+        };
+      });
+
+      Object.entries(healthSummary).forEach(([key, item]) => {
+        healthSummary[key].healthy =
+          or0(item.successful) + or0(item.clean_shutdown) + or0(item.running);
+        healthSummary[key].unhealthy =
+          or0(item.failed) +
+          or0(item.spot_kill) +
+          or0(item.insufficient_capacity) +
+          or0(item.volume_limit_exceeded) +
+          or0(item.missing_ami) +
+          or0(item.startup_failed) +
+          or0(item.unknown_codes) +
+          or0(item.no_code);
+      });
+
+      const sortByProperty = camelCase(sortBy);
+      const healthValues = Object.values(healthSummary);
+
+      if (!sortBy) {
+        return healthValues;
+      }
+
+      return ifElse(
+        isEmpty,
+        identity,
+        rSort((a, b) => {
+          const firstElement =
+            sortDirection === 'desc' ? b[sortByProperty] : a[sortByProperty];
+          const secondElement =
+            sortDirection === 'desc' ? a[sortByProperty] : b[sortByProperty];
+
+          return sort(firstElement, secondElement);
+        })
+      )(healthValues);
+    }
+  );
+
+  handleDrawerOpen = (healthItem, columnName) => {
+    this.setState({
+      drawerOpen: true,
+      drawerItem: {
+        ...healthItem,
+        columnName,
+      },
+    });
+  };
+
+  handleDrawerClose = () => {
+    this.setState({
+      drawerOpen: false,
+      drawerItem: null,
+    });
+  };
+
+  render() {
+    const { classes, healthData } = this.props;
+    const { sortBy, sortDirection, drawerOpen, drawerItem } = this.state;
+    const sortedHealth = this.createSortedHealth(
+      healthData,
+      sortBy,
+      sortDirection
+    );
+    const iconSize = 18;
```

Comment:
This can be a constant at the top since it doesn't need to change with each render.